### PR TITLE
Provide option for msbuild properties

### PR DESCRIPTION
**Pull request by vicancy**
for #1518 & #1254

### CODE REVIEW

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Incremental/CacheBase.cs [hunk 1]
```diff
@@ -38,7 +38,7 @@ public BuildInfo GetValidConfig(IEnumerable<string> inputProjects)
             return GetConfig(key);
         }
 
-        public void SaveToCache(IEnumerable<string> inputProjects, IDictionary<string, List<string>> containedFiles, DateTime triggeredTime, string outputFolder, IList<string> fileRelativePaths, bool shouldSkipMarkup)
+        public void SaveToCache(IEnumerable<string> inputProjects, IDictionary<string, List<string>> containedFiles, DateTime triggeredTime, string outputFolder, IList<string> fileRelativePaths, bool shouldSkipMarkup, IDictionary<string, string> msbuildProperties)
```
Comment (vwxyzh) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Incremental/CacheBase.cs]: create a class for parameters?

**Comment by vwxyzh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/Incremental/CacheBase.cs (hunk 1 above)
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/Incremental/CacheBase.cs]: will do refactor later

**Comment by ericdrobinson** [comment]
Does this Issue perchance cover preprocessor symbol handling as suggested in #1518?

>or at least specify C# preprocessor symbols to define?

We have some APIs that are only generated for certain builds where certain Preprocessor Symbols are defined. It would be awesome if the same symbols specified when building assemblies could similarly affect the generated documentation (i.e. make generated documentation content match the generated code content)...

**Comment by macrogreg** [comment]
The `merge` command described in the release notes for this PR is not actually working (https://github.com/dotnet/docfx/issues/3067, https://github.com/dotnet/docfx/issues/7106).

**Pull request merged**